x-pack/plugin/esql/qa/testFixtures/src/main/resources/mv_expand.csv-spec

```diff
@@ -225,3 +225,11 @@ emp_no:integer |  job_positions:keyword
 10004          |Head Human Resources
 10004          |Reporting Analyst
 ;
+
+
+expandFoldable
+row a = "foobar", b = ["foo", "bar"], c = 12 | mv_expand b | where b LIKE "fo*";
```

Comment:
Is this the only scenario where this PR applies?
What about `from test | sort emp_no | where emp_no <10050 | limit 10 | eval foldable = "foo,bar" | eval generate_mv = split(foldable,",") | mv_expand generate_mv | keep emp_no, first_name, generate_mv`?

Reply:
Right, eval can also create foldable expressions, let me add some tests

Reply:
That particular query fails (`Block does not have same position count`), but the problem is not related to this PR, it also fails in `main`.

Actually, the planning of that query is not influenced by this fix, because there are no other operations after MV_EXPAND (eg, a WHERE) that could receive wrong input information.

Even worse, the problem seems to be due to EVAL, not to MV_EXPAND; also the following fails with the same error

```
from employees | sort emp_no | where emp_no <10050 | limit 10 | eval foldable = \"foo,bar\" | eval generate_mv = split(foldable,\",\")
```

I'll open a separate issue.

Unfortunately this is kind of a blocker for testing this PR thoroughly, so probably I'll have to address that problem first

Reply:
Thank you for creating a separate issue. I didn't intend to block this PR with that specific query suggestion (haven't tested it upfront). I suggest putting this one in first and then address the other one. And when looking into the `eval` issue you can add more tests that will cover `mv_expand` in non-`row` scenarios.

Reply:
https://github.com/elastic/elasticsearch/issues/101425